starter-code/src/memory.js

```diff
@@ -1,15 +1,31 @@
-// var MemoryGame = function (cards) {
-//   this.cards = cards;
-// };
+var MemoryGame = function(cards) {
+  this.cards = cards;
+  this.pickedCards = [];
+  this.pairsClicked = 0;
+  this.pairsGuessed = 0;
+};
 
-// MemoryGame.prototype.shuffleCard = function (cardsArr) {
+MemoryGame.prototype.shuffleCard = function(cardsArr) {
+  for (var i = cardsArr.length - 1; i > 0; i--) {
+    var j = Math.floor(Math.random() * (i + 1));
+    [cardsArr[i], cardsArr[j]] = [cardsArr[j], cardsArr[i]];
+  }
+  return cardsArr;
+};
 
-// };
-
-// MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
-
-// }
-
-// MemoryGame.prototype.finished = function () {
-
-// };
+MemoryGame.prototype.checkIfPair = function(firstCard, secondCard) {
+  this.pairsClicked += 1;
+  if (firstCard === secondCard) {
+    this.pairsGuessed += 1;
+    return true;
+  } else {
+    return false;
+  }
+};
+MemoryGame.prototype.finished = function() {
+  if (this.pairsGuessed === 12) {
```

Comment:
A more generic version would be 
`if (this.pairsGuessed === this.cards.length/2)`